webapp/src/ts/services/contact-stats.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import * as moment from 'moment';
+import { isObject as _isObject, uniq as _uniq } from 'lodash-es';
+import * as CalendarInterval from '@medic/calendar-interval';
+
+import { DbService } from '@mm-services/db.service';
+import { SessionService } from '@mm-services/session.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ContactStatsService {
+
+  constructor(
+    private dbService: DbService,
+    private sessionService: SessionService,
+  ) { }
+
+  private async getLastVisitedDate(contactId) {
+    const queryParams: any = { reduce: true, group: true };
+
+    if (this.sessionService.isOnlineOnly()) {
+      // Querying with keys in PouchDB is not optimal, so adding key for online users only.
+      queryParams.key = contactId;
+    }
+
+    const records = await this.dbService
+      .get()
+      .query('medic-client/contacts_by_last_visited', queryParams);
+
+    return records.rows.find(row => contactId === row.key);
+  }
+
+  private async getVisitsInDateRange(dateRange: DateRange, contactId) {
```

Comment:
I was wondering about this. 
For this kind of change in the views, do we need to consider backward compatibility? 🤔

Reply:
We would if the view is used by other services, buy I don't think this view is. Definitely check!

Reply:
What I was thinking is that we could change the view to emit a new key along with the key it's already emitting. 
This can be a composite key with `[visited_contact_uuid, date]`, so that way you can query with a start and end key for a single contact, and the query that already uses this view in search shouldn't be affected.

Reply:
done